src/app/pages/receive/components/receive-item.tsx

```diff
@@ -24,23 +25,33 @@ export function ReceiveItem({
 }: ReceiveItemProps) {
   if (!address) return null;
   return (
-    <Flag align="middle" img={icon} spacing="space.04">
-      <Flex justifyContent="space-between">
-        <Stack gap="space.00">
-          <styled.span textStyle="label.02">{title}</styled.span>
-          <styled.span textStyle="caption.02">{truncateMiddle(address, 6)}</styled.span>
-        </Stack>
-        <HStack gap="space.00">
-          <Button onClick={onCopyAddress} variant="ghost">
-            <CopyIcon />
-          </Button>
-          {onClickQrCode && (
-            <Button data-testid={dataTestId} ml="space.02" onClick={onClickQrCode} variant="ghost">
-              <QrCodeIcon />
-            </Button>
-          )}
-        </HStack>
-      </Flex>
-    </Flag>
+    <Item.Root>
+      <Item.Content>
+        <ItemWithButtonsLayout
+          flagImg={icon}
+          contentLeftTop={<Item.Text>{title}</Item.Text>}
+          contentLeftBottom={
+            <Item.Caption textStyle="caption.01">{truncateMiddle(address, 6)}</Item.Caption>
```

Comment:
How does `Item.Caption` differ from `Caption`?

I'd say either: 
- If we want to force caption, put it in the component per default and all content of `contentLeftBottom` goes inside a `<Caption />`
- or, we want it entirely custom, in which case we use `<Caption />` in all cases and don't create another in the form of `Item.Caption`

Reply:
I think our goal was to remove the `Caption` component? I thought we created it during the UI refactor just temporarily? I think it makes more sense to style the `Item.Caption` as a slot here with the item.

Reply:
Also here, I thought the point of using `slots` was so that the slot components could use variant styling ...like for disabled? I think that is why I thought easier to include so that no mater what layout it is used for, then the contents would get those disabled styles.

Reply:
Then just `<styled.span textStyle="" />`? 

Slots have a few uses, like sharing state and encapsulating functionality within a component instance. Really enjoyed using `sva` for the Input, but less sure this is the right use case for it.

I feel like we shouldn't build this in a way where we say "can be a button or not" with `isPressed`, because to do that, we're rebuilding the web, adding keyboard events and disabled states to divs, rather than just using a button, and handling non-interactive elements differently. It's complexity we can avoid, I think. We don't need to keep using isPressed just because we had it before (even if it is more work to remove).

For me, the core problem here is that we are coupling two concerns that should be separate: 
1) the styling for generic interactive elements
  e.g. these boxes 
<img width="250" alt="image" src="https://github.com/leather-wallet/extension/assets/1618764/3abf0260-7dc2-407e-a9fd-fd35004c9dfc">

2) styling of content that commonly used within one instance of an interactive element
  e.g. these states 
<img width="350" alt="image" src="https://github.com/leather-wallet/extension/assets/1618764/355c9c7b-2504-4447-9acd-6b3b93829e5b">

To build the simple instance in no. 1 with this implementation, we're carrying around a `Item.Caption` component, even though it isn't used.

Do you think it could work this way?

```ts
const defaultItemStyles = css({ ... });

const interactiveStyles = css({
   ...defaultItemStyles,
  ...allTheAdditionalHoverFocusDisabledEtcStates,
});

function NonInteractiveRowCompoent () {
  // other jsx, styles
  return <styled.span className={defaultItemStyles} />;
}

function CompoentThatLooksLikeExampleTwo () {
  // this component defines all the captions and complex layouts of the 
  return <styled.button className={interactiveStyles} />;
}
```

Reply:
Ok, thanks for the feedback, I'll try a simpler version like this in another commit and see what we think. 👍